pkg/controller/daemon/controller_test.go

```diff
@@ -200,17 +201,127 @@ func TestNotReadNodeDaemonDoesNotLaunchPod(t *testing.T) {
 	syncAndValidateDaemonSets(t, manager, ds, podControl, 0, 0)
 }
 
-// DaemonSets should not place onto Unschedulable nodes
-func TestUnschedulableNodeDaemonDoesNotLaunchPod(t *testing.T) {
+// DaemonSets should not place onto OutOfDisk nodes
+func TestOutOfDiskNodeDaemonDoesNotLaunchPod(t *testing.T) {
 	manager, podControl := newTestController()
-	node := newNode("not-ready", nil)
-	node.Spec.Unschedulable = true
+	node := newNode("not-enough-disk", nil)
+	node.Status.Conditions = []api.NodeCondition{{Type: api.NodeOutOfDisk, Status: api.ConditionTrue}}
+	manager.nodeStore.Add(node)
+	ds := newDaemonSet("foo")
+	manager.dsStore.Add(ds)
+	syncAndValidateDaemonSets(t, manager, ds, podControl, 0, 0)
+}
+
+// DaemonSets should not place onto nodes with insufficient capacity
```

Comment:
Oh, actually capacity == free rsources because there isn't any other pod on the node. Can you add a pod to the node at the beginning of the test, such that the daemon's requirements < the node's capacity, but the daemon's requirements > the node's free space, thus daemon doesn't schedule? (You can reuse the same test, just make it a little more complicated by adding a pod.)